### PR TITLE
Max pool2d ceil mode to tosa

### PR DESCRIPTION
**Pull request by lisaliu1**
this PR implements the support for lowering torch.aten.max_pool2d to tosa with ceil_mode==true.

### CODE REVIEW

**Review by ljfitz — APPROVED** [review]
Looks good to me. It'd be good to check if the other backends to support the mode, as otherwise you'll get failures in testing during the PR. (Perhaps you could turn on workflow for your fork if you want to see results early, but I confess I've never tried.)

**Comment by lisaliu1** [comment]
I created the PR to the upstream. So far, 4 out 5 checks have passed.
https://github.com/llvm/torch-mlir/pull/1957

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -3793,14 +3793,24 @@ class ConvertAtenPoolingBaseOp : public OpConversionPattern<AtenOpT> {
...
+      }
+      else {
+        if (dimSize % stride == 0) {
+          dimSize = (dimSize / stride + 1);
+        }
+        else {
+          dimSize = (dimSize / stride + 2);
+        }
+      }
+      return dimSize;
```
Comment (ljfitz) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:3813]: I somehow lost this comment - not sure where it went.
```suggestion
      int64_t dimSize = inputDim + padBefore + padAfter - dilation * (kernelDim - 1) - 1;
      if (ceilMode && (dimSize % stride != 0))
        return dimSize / stride + 2;
      return dimSize / stride + 1;
```

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 2]
```diff
@@ -3793,14 +3793,24 @@ class ConvertAtenPoolingBaseOp : public OpConversionPattern<AtenOpT> {
 
   static int64_t getOutputDim(int64_t inputDim, int64_t kernelDim,
                               int64_t stride, int64_t padBefore,
-                              int64_t padAfter, int64_t dilation) {
+                              int64_t padAfter, int64_t dilation,
+                              bool ceilMode=false) {
     if (inputDim == kUnknownSize) {
       return kUnknownSize;
```
Comment (ljfitz) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:3799]: I also lost a comment saying that this wasn't you're code, but perhaps good to adjust to the guidelines:
- https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return
- https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

Comment (lisaliu1) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:3799]: Thanks, that's very helpful. I got reviews regarding the coding style as well.

**Pull request closed without merging**